endpoints/swagger_generator.py

```diff
@@ -837,8 +837,6 @@ def get_descriptor_defaults(self, api_info, hostname=None):
                 api_info.hostname)
     protocol = ('http' if hostname and hostname.startswith('localhost') else
                 'https')
```

Comment:
Yep. I also wrote some new tests for swagger generation to make sure the protocol is showing up correctly in the devserver case there, along with some refactoring for the test. PTAL.